src/bin/server.rs

```diff
@@ -21,6 +21,19 @@ fn main() -> Result {
         return static_web_server::settings::cli_output::display_version();
     }
 
+    if let Ok(path) = std::env::var("SWS_GENERATE") {
```

Comment:
Dumb question, is this feature intended for Linux/Unix(es) only? How about Windows? If it is only for *nixes then we could provide it under a compile-time `cfg` macro.

Reply:
I'm not sure whether Windows has man pages, but at least the completions include completions for powershell, so at least that bit is not specific to the *nixes

Reply:
Good, then let's make man pages available only for *nixes and completions for all OSes.

Reply:
not supporting man page generation on windows makes this quite a bit more complicated I think. How important is that to you?

Reply:
My point was if man pages are not supported in Windows then we can disable it. Otherwise, if supported then let's leave it like you said.

Reply:
In the end, a man page is just a text file right? Does not matter if windows is able to read it or not. 😅

Reply:
Yeah, it's just generating a text file that might not be readable by any installed software. Some people do install groff (the format behind man pages) readers on Windows as well, so it really doesn't matter.

This is the last thing where I wasn't sure, the rest should be good to go I think.

I haven't tested compilation on Windows yet, but from my reading of the code it *should* work. If you want I can look into getting a VM to try it out.

If you're happy with man pages being generated for windows as well, then this is ready for review now.

Reply:
I didn't test on Windows yet but it should work.